### PR TITLE
WIP: Conan center hook

### PR DESCRIPTION
**Pull request by lasote**
- Renamed to conan-center.
- Simplified code by using decorators managing the output
- Introduced a way to fail when an error, by an environment variable.
- To discuss if the decorators might have sense in other generators (provide them from conan?).
- Tested with c3i

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,11 @@ To install all hooks from Conan repository in Github:
 
 ``$ conan config install https://github.com/conan-io/hooks``
 
+If you are using Conan >=1.14 you can specify the source and destination folder to avoid copying
```
Comment (uilianries) [inline README.md]: nice!

File: hooks/conan-center.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
+import fnmatch
```
Comment (uilianries) [inline hooks/conan-center.py]: you should have used `git mv` :sweat:

File: hooks/conan-center.py [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+    return tmp
+
+
+@raise_if_error_output
+def pre_export(output, conanfile, conanfile_path, reference, **kwargs):
+    conanfile_content = tools.load(conanfile_path)
+    settings = getattr(conanfile, "settings", None)
+
+    @run_test("REFERENCE LOWERCASE", output)
+    def test(out):
```
Comment (uilianries) [inline hooks/conan-center.py]: is it correct, all methods have the same name `test`

Comment (lasote) [reply hooks/conan-center.py]: the name is irrelevant

File: hooks/conan-center.py (hunk 2 above)
Comment (lasote) [reply hooks/conan-center.py]: not sure if even using git mv it works if the diff is > X%

File: hooks/conan-center.py (hunk 3 above)
Comment (uilianries) [reply hooks/conan-center.py]: I see you are using a wrapper, but my eyes hurt when I see duplicate names :joy_cat:

File: hooks/conan-center.py (hunk 2 above)
Comment (uilianries) [reply hooks/conan-center.py]: Yes, it's a totally new hook :heart_eyes:

File: hooks/conan-center.py (hunk 3 above)
Comment (jgsogo) [reply hooks/conan-center.py]: I like the implementation, it's cleaner, but I would make each `test` a free function and then call them from hook's methods, I think it is better that implicitly calling the decorated function from the decorator itself (not tested, but I have some doubts about it):

```python
def reference_lowercase(output, reference, **kwargs)
    ...

@raise_if_error_output
def pre_export(*args, **kwargs):
    run_test("REFERENCE LOWERCASE", reference_lowercase, **kwargs)
```

**Comment by danimtb** [comment]
A fix for the tests already in place would be needed

**Comment by lasote** [comment]
I don't want to merge it yet actually. we have to validate if it covers every rule or restriction we want for the new ccci

**Comment by danimtb** [comment]
Please update this PR with the changes in Master branch so both Travis and Appveyor CIs pass the tests. We fixed some issues last week.

**Comment by danimtb** [comment]
Now there are some tests failing due to new checks introduced recently

**Pull request merged**